endpoints/querier/querier.go

```diff
@@ -14,45 +12,34 @@ var netClient = &http.Client{
 	Timeout: time.Second * 35,
 }
 
-// GetResponseAndTiming returns the http response, the reponse time and any errors for an http request to the endpoint at urlString
-func GetResponseAndTiming(urlString string) (*http.Response, float64, error) {
+// GetResponseAndTiming returns the http response, the reponse time, the context cancel function and any errors for an http request to the endpoint at urlString
+func GetResponseAndTiming(urlString string) (*http.Response, float64, context.CancelFunc, error) {
```

Comment:
from the bit I've read, I think it makes more sense to return the straight context rather than `context.CancelFunc`. The receiving function can then call `context.CancelFunc` if needed but could call other things as well.

@ttaylor249

Reply:
Nevermind? It looks like the cancel func is not directly a part of the context? still would like @ttaylor249's thoughts.